wifi: Properly handle users making mistakes with the Wi-Fi key

bochecha opened the pull request:
NetworkManager allows creating a new connection either for just the current user, or shared with all other users on the system.

Before this commit, we were doing the latter, which requires additional permissions.

Since we only really care about the ideascube user, let's only create a connection for that user.

yohanboniface commented on ideascube/serveradmin/wifi.py:
```diff
@@ -72,6 +72,7 @@ def _new_connection(self, wifi_key=None):
                 'id': self.ssid,
                 'type': '802-11-wireless',
                 'uuid': str(uuid4()),
+                'permissions': ['user:mathieu:'],
```
mathieu? :)

yohanboniface commented on ideascube/serveradmin/wifi.py:
```diff
@@ -97,7 +103,20 @@ def _new_connection(self, wifi_key=None):
+            keyprop = '802-11-wireless-security.psk:'
+
+            if excname == name and excmsg.startswith(keyprop):
+                raise WifiError(_('Invalid Wi-Fi key'))
```
Can you add a minimal unittest for that case?

bochecha commented:
So I went back on this one, and removed the code that was creating per-user connections.

When storing per-user connections, NetworkManager wants an actual user session (login).

Given that out target for deployment is a system-level service, we don't have user sessions.

Storing a system-level session also means that the connection is activated at boot-time, rather than when the user logs in.

All in all, the change I had made which was storing per-user connections was wrong, and I now just removed it from this pull request.

The other one still holds though: we can't just return a 500 error when the user sends an invalid Wi-Fi passphrase.

I only added unit tests for it as you requested @yohanboniface.
